test/testcafe/spec/ChallengeAuthenticatorRsa_spec.js

```diff
@@ -0,0 +1,92 @@
+import { RequestMock } from 'testcafe';
+import xhrAuthenticatorRequiredRsa from '../../../playground/mocks/data/idp/idx/authenticator-verification-rsa';
+import xhrInvalidPasscode from '../../../playground/mocks/data/idp/idx/error-authenticator-verification-rsa';
+import xhrPasscodeChange from '../../../playground/mocks/data/idp/idx/error-authenticator-verification-passcode-change-rsa';
+import xhrSuccess from '../../../playground/mocks/data/idp/idx/success';
+import ChallengeRsaPageObject from '../framework/page-objects/ChallengeOnPremPageObject';
+import SuccessPageObject from '../framework/page-objects/SuccessPageObject';
+
+const mockChallengeAuthenticatorRsa = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrAuthenticatorRequiredRsa)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(xhrSuccess);
+
+const mockInvalidPasscode = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrAuthenticatorRequiredRsa)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(xhrInvalidPasscode, 403);
+
+const mockPasscodeChange = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrAuthenticatorRequiredRsa)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/answer')
+  .respond(xhrPasscodeChange, 403);
+
+fixture('Challenge Authenticator RSA');
+
+async function setup(t) {
+  const challengeRsaPage = new ChallengeRsaPageObject(t);
+  await challengeRsaPage.navigateToPage();
+  return challengeRsaPage;
+}
+
+test.requestHooks(mockChallengeAuthenticatorRsa)('challenge RSA authenticator', async t => {
+  const challengeRsaPage = await setup(t);
+
+  const { log } = await t.getBrowserConsoleMessages();
```

Comment:
Just FYI of an upcoming change: @jmelberg-okta  has a PR out (https://github.com/okta/okta-signin-widget/pull/1658) to remove all this duplicate assertions of console logs! Once that is merged, you won't have to do this again